### ISSUE TITLE
Proposal: error when referencing packages with the same name

### ISSUE THREAD

**Issue by guibou** [tweag/rules_haskell#402]
Bug #395 shows that when two packages with the same bazel rule `name` are used as dependencies of the same rule, it may results in unexpected errors.

Example of usage:

```python
haskell_library(
   name = "aLib",
   deps = [
       "@repo//foo/bar:baz", # rule name: baz
       "@repo2//spam/eggs:baz", # rule name: baz
   ],
)
```

Errors:

- link issue (only one package is linked). Discussion in #396 shows that it may be a GHC bug.
- macro conflict (both packages expose `MIN_VERSION_pkgname` and `VERSION_pkgname` CPP macros). This is the documented behavior of GHC which exports these variables since GHC 8.0.

Discussion in #396 shows that any robust solution, involving a change in package naming, may have impact on the ecosystem, such as hazel. Such a solution must be discussed.

In the meantime, users are facing unexpected issues without any precise error reporting which may point them to their module naming. There is a simple workaround for them: rename their dependencies, but they need to be aware of the solution.

That's the reason why I propose to introduce a new *error* in `rules_haskell` when a rule reference two dependencies with the same package name.

I'll work on a pull request if this proposal is accepted.

**Comment by guibou** [comment]
# Discussion

An *error* should not break any existing code (because they were already broken) and if by any chance it happen, the fix will be obvious for the user.
We can also set it as a *warning*, this way it won't break anything but will still provide the mandatory information.

**Comment by Profpatsch** [comment]
I think it should throw an error, at least as long as GHC [silently ignores packages given by ID that have the same name as an other package](  GHC upstream issue about https://github.com/tweag/rules_haskell/pull/396#issuecomment-412040610).

**Comment by guibou** [comment]
@m fixed the most annoying issue in #403 so an error is not necessary now.

And a warning will be a lot of noise, so for now, I'm closing this PR.
